src/render/dom/HTMLVisualElement.ts

```diff
@@ -423,39 +414,46 @@ export class HTMLVisualElement<
         eachAxis(axis => this.axisProgress[axis].stop())
     }
 
-    /**
-     * Update the layout deltas to reflect the relative positions of the layout
-     * and the desired target box
-     */
-    updateLayoutDeltas(isReactRender: boolean) {
-        /**
-         * Ensure that all the parent deltas are up-to-date before calculating this delta.
-         *
-         * TODO: This approach is exceptionally wasteful as every child will update
-         * the deltas of its parent even if it's already updated for this frame.
-         * We can optimise this by replacing this to a call directly to the root VisualElement
-         * which then runs iteration from the top-down, but only once per framestamp.
-         */
-        this.treePath.forEach((p: HTMLVisualElement) =>
-            p.updateLayoutDeltas(isReactRender)
-        )
+    updateLayoutDelta = () => {
+        this.isLayoutProjectionEnabled && this.updateDeltas()
 
         /**
-         * Early return if layout reprojection isn't enabled
+         * Ensure all children layouts are also updated.
+         *
+         * This uses a pre-bound function executor rather than a lamda to avoid creating a new function
+         * multiple times per frame (source of mid-animation GC)
          */
-        if (!this.isLayoutProjectionEnabled || !this.box) return
+        this.children.forEach(fireUpdateLayoutDelta)
+    }
 
+    /**
+     * Update the layout deltas to reflect the relative positions of the layout
+     * and the desired target box
+     */
+    updateDeltas() {
         /**
          * Reset the corrected box with the latest values from box, as we're then going
          * to perform mutative operations on it.
          */
         resetBox(this.boxCorrected, this.box)
 
+        /**
+         * If this component has a parent, update this treeScale by incorporating the parent's
+         * delta into its treeScale.
+         */
+        if (this.parent) {
+            updateTreeScale(
+                this.treeScale,
+                (this.parent as any).treeScale,
+                (this.parent as any).delta
+            )
+        }
+
         /**
          * Apply all the parent deltas to this box to produce the corrected box. This
          * is the layout box, as it will appear on screen as a result of the transforms of its parents.
          */
-        applyTreeDeltas(this.boxCorrected, this.treeScale, this.treePath as any)
+        applyTreeDeltas(this.boxCorrected, this.treePath as any)
```

Comment:
Because we've already updated parent deltas we can simply use the latest treeScale value in updateTreeScale rather than traversing the tree a second time. We still do run this loop but we're doing less work per iteration as a result.